### PR TITLE
Fix linear decay for learning rate in `Doc2Vec.infer_vector`. Fix #2061

### LINKED ISSUE #2061: Doc2Vec.infer_vector learning rate decays extremely fast (non-linearly)
I am working with a corpus of very short documents and noticed that the inferred vectors for the same document were very different.

```python
from scipy.spatial.distance import pdist, squareform
testdoc = "This is a small sample document."
vectors = [d2vmod.infer_vector(testdoc) for _ in range(5)]
squareform(pdist(vectors, "cosine"))
```

```
array([[0.        , 0.05987812, 0.06183155, 0.06931093, 0.05466599],
       [0.05987812, 0.        , 0.03724874, 0.05006329, 0.04789369],
       [0.06183155, 0.03724874, 0.        , 0.04771786, 0.05983109],
       [0.06931093, 0.05006329, 0.04771786, 0.        , 0.0367826 ],
       [0.05466599, 0.04789369, 0.05983109, 0.0367826 , 0.        ]])
```

More training steps makes things worse in this case:
```python
vectors = [d2vmod.infer_vector(testdoc, 10000) for _ in range(5)]
squareform(pdist(vectors, "cosine"))
```
```
array([[0.        , 0.27392197, 0.308742  , 0.51374501, 0.45744246],
       [0.27392197, 0.        , 0.14912033, 0.32902151, 0.1822687 ],
       [0.308742  , 0.14912033, 0.        , 0.2895444 , 0.27019636],
       [0.51374501, 0.32902151, 0.2895444 , 0.        , 0.38096254],
       [0.45744246, 0.1822687 , 0.27019636, 0.38096254, 0.        ]])
```
Note: This is more extreme than what I'm seeing with more domain-specific sample documents, where start to get more consistent after about 5000 steps.

I believe this is happening because the learning rate decays extremely rapidly:
 https://github.com/RaRe-Technologies/gensim/blob/8b810918d59781116794a6679999afdc76b857ef/gensim/models/doc2vec.py#L565

```python
alpha = 0.025
min_alpha = 0.001
steps = 100
for i in range(steps):
    print(alpha)
    alpha = ((alpha - min_alpha) / (steps - i)) + min_alpha
```
```
0.025
0.00124
0.0010024242424242424
0.0010000247371675943
...
```
Notice that `alpha` is very close to `min_alpha` after the first step and this is exaggerated even more when the number of steps is larger. 

When I change Doc2Vec to have a linear decay in learning rate
```python
alpha_delta = (alpha-min_alpha)/(steps-1)
for i in range(steps):
    # ...
    alpha -= alpha_delta
```

I get much better results. With 20 steps, we get pairwise cosine distances of
```
array([[0.        , 0.01617053, 0.02467067, 0.01828433, 0.01834735],
       [0.01617053, 0.        , 0.01879757, 0.00910884, 0.01358116],
       [0.02467067, 0.01879757, 0.        , 0.01521225, 0.01392789],
       [0.01828433, 0.00910884, 0.01521225, 0.        , 0.01121792],
       [0.01834735, 0.01358116, 0.01392789, 0.01121792, 0.        ]])
```
, with 100 we get
```
array([[0.        , 0.00282428, 0.00373375, 0.00331408, 0.00362875],
       [0.00282428, 0.        , 0.0036147 , 0.0028999 , 0.00210812],
       [0.00373375, 0.0036147 , 0.        , 0.0032986 , 0.00361321],
       [0.00331408, 0.0028999 , 0.0032986 , 0.        , 0.00318849],
       [0.00362875, 0.00210812, 0.00361321, 0.00318849, 0.        ]])
```
, and with 1000 steps:
```
array([[0.        , 0.00055459, 0.000633  , 0.00074271, 0.00036596],
       [0.00055459, 0.        , 0.00067211, 0.00075522, 0.00058975],
       [0.000633  , 0.00067211, 0.        , 0.00109709, 0.00049239],
       [0.00074271, 0.00075522, 0.00109709, 0.        , 0.00072527],
       [0.00036596, 0.00058975, 0.00049239, 0.00072527, 0.        ]])
```

### PR DESCRIPTION
**Pull request by umangv**
This would fix issue #2061

### CODE REVIEW

**Comment by gojomo** [comment]
Looks good to me! This should be a high-priority fix. @menshikh-iv

**Comment by piskvorky** [comment]
Any updates needed to tutorial / docs, after this change?

**Comment by gojomo** [comment]
This is always how it should have worked. If tests are passing (and they are), I'd expect any uses of inference in tutorials to just start working better – maybe a little, maybe a lot. (I wouldn't make merging this tiny fix dependent on refreshing all those result-cells.)

**Comment by menshikh-iv** [comment]
Thanks @umangv, congratz with the first contribution :1st_place_medal:

**Comment by piskvorky** [comment]
@gojomo @umangv can you draft a summary of this PR for this changelog? From the user perspective (not internal dev): who is affected, in what way, what should they do about it (if anything).

High-level, a few short sentences.

**Comment by umangv** [comment]
This change fixes a bug in `Doc2Vec.infer_vector` that caused the learning rate to decay very fast. With this change, `infer_vectors` method should return more consistent document vectors and require far fewer steps to converge, especially for short documents.
